Yafc/Windows/MainScreen.cs

```diff
@@ -95,23 +95,27 @@ private void SetProject(Project project) {
             _ = InputSystem.Instance.SetDefaultKeyboardFocus(this);
         }
 
-        private void ProjectSettingsChanged(bool visualOnly) {
+        private async void ProjectSettingsChanged(bool visualOnly) {
             if (visualOnly) {
                 return;
             }
 
             if (topScreen == null) {
-                ReRunAnalysis();
+                await ReRunAnalysis();
             }
             else {
                 analysisUpdatePending = true;
             }
         }
 
-        private void ReRunAnalysis() {
+        private async Task ReRunAnalysis() {
             analysisUpdatePending = false;
             ErrorCollector collector = new ErrorCollector();
             Analysis.ProcessAnalyses(this, project, collector);
+            project.pages.ForEach(p => p.SetToRecalculate());
+            Task primary = (activePage?.RunSolveJob() ?? Task.CompletedTask);
+            await (secondaryPage?.RunSolveJob() ?? Task.CompletedTask);
+            await primary;
```

Comment:
Could go with `Task.WaitAll` here to wait for both jobs. That way if the second one fails, you get thrown out of it instantly.